api/hypershift/v1beta1/hosted_controlplane.go

```diff
@@ -313,6 +323,12 @@ type HostedControlPlaneStatus struct {
 	// for this control plane.
 	KubeConfig *KubeconfigSecretRef `json:"kubeConfig,omitempty"`
 
+	// customKubeConfig specifies the name and key for the External Custom kubeconfig secret.
```

Comment:
This doc sounds like this is an spec field but it's a status one.
>When set, they result in the generation of a secret with the given name containing a Kubeconfig within the hostedCluster namespace and a referenced by the hostedCluster.status object.

This field doesn't cause a generation of a secret, this field is the result of generating that secret, right?